controllers/numaresourcesoperator_controller.go

```diff
@@ -73,6 +73,12 @@ import (
 
 const numaResourcesRetryPeriod = 1 * time.Minute
 
+// poolDaemonSet a struct to hold the target MCP of a configured node group and its created respective RTE daemonset
+type poolDaemonSet struct {
+	PoolName  string
+	DaemonSet nropv1.NamespacedName
```

Comment:
for the future: let's add a methof in the `api/v1` package:
```
func (nn NamespacedName) AsKey() client.ObjectKey {
	return client.ObjectKey{
		Namespace: nn.Namespace,
		Name:      nn.Name,
	}
}
```

Reply:
actually bad idea, we will bring the client package as dep into our api package, which is something we don't want.

Reply:
ack

Reply:
*in another PR* we can add another helper package though

Reply:
PR: https://github.com/openshift-kni/numaresources-operator/pull/1053